security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -148,9 +161,11 @@ func (sc *SecretCache) rotate(t time.Time) {
 		if sc.shouldRefresh(&e) {
 			go func() {
 				if sc.isTokenExpired(&e) {
-					// Send the notification to close the stream connection if both cert and token have expired.
-					if err := sds.NotifyProxy(proxyID, nil /*nil indicates close the streaming connection to proxy*/); err != nil {
-						log.Errorf("Failed to notify for proxy %q: %v", proxyID, err)
+					if sc.notifyCallback != nil {
```

Comment:
Is it possible for this callback be nil? If yes, should we handle the nil case like logging/returning some error, otherwise we don't need this check.

Reply:
added log, it's safer to add the check to prevent unexpected nil reference crash nodeagent.

Reply:
sg